### PR TITLE
 [core] Removed all `NativeModulesProxy` occurrences

### PR DESCRIPTION
**Pull request by reichhartd**
# Why

In order to remove `NativeModulesProxy` from `expo-modules-core` in the future, I have replaced the use of it in all other packages.

# How

I have replaced all occurrences of the deprecated `NativeModulesProxy`.

I have left the export of `NativeModulesProxy` in the `index.ts` of `expo-modules-core`, but can remove it if it is not too early.

# Test Plan

Test cases run.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-modules-core/CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +84,7 @@
 - [Android] Cached argument's type information to improve memory and startup time. ([#31297](https://github.com/expo/expo/pull/31297) by [@lukmccall](https://github.com/lukmccall))
 - [iOS] Make `func application(_ application: UIApplication, supportedInterfaceOrientationsFor window: UIWindow?)` `open`. ([#31398](https://github.com/expo/expo/pull/31398) by [@abulenok](https://github.com/abulenok))
 - [Android] Expose `RuntimeContext.eval()` as `JavaScriptRuntime.eval()` on iOS. ([#31445](https://github.com/expo/expo/pull/31445) by [@kudo](https://github.com/kudo))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-modules-core/CHANGELOG.md:87]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-screen-orientation/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@
 ### 💡 Others
 
 - Removed redundant usage of `EventEmitter` instance. ([#28946](https://github.com/expo/expo/pull/28946) by [@tsapeta](https://github.com/tsapeta))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-screen-orientation/CHANGELOG.md:18]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-secure-store/CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@
 ### 💡 Others
 
 - [iOS] check return value of SecAccessControlCreateWithFlags ([#29983](https://github.com/expo/expo/pull/29983) by [@vonovak](https://github.com/vonovak))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-secure-store/CHANGELOG.md:19]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-sms/CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,8 @@
 
 ### 💡 Others
 
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-sms/CHANGELOG.md:15]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-constants/CHANGELOG.md [hunk 5]
```diff
@@ -16,6 +16,7 @@
 ### 💡 Others
 
 - Deprecate `Constants.appOwnership`. ([#30021](https://github.com/expo/expo/pull/30021) by [@amandeepmittal](https://github.com/amandeepmittal))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-constants/CHANGELOG.md:19]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-font/CHANGELOG.md [hunk 6]
```diff
@@ -23,6 +23,7 @@
 
 - Stopped scoping font family names in Expo Go on Android. ([#28797](https://github.com/expo/expo/pull/28797) by [@lukmccall](https://github.com/lukmccall))
 - Replaced `@testing-library/react-hooks` with `@testing-library/react-native`. ([#30742](https://github.com/expo/expo/pull/30742) by [@byCedric](https://github.com/byCedric))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-font/CHANGELOG.md:26]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

File: packages/expo-maps/CHANGELOG.md [hunk 7]
```diff
@@ -11,13 +11,14 @@
...
 - Removed usage of reserved keyword `module`. ([#30010](https://github.com/expo/expo/pull/30010) by [@EvanBacon](https://github.com/EvanBacon))
-Add missing `react` and `react-native` peer dependencies for isolated modules. ([#30475](https://github.com/expo/expo/pull/30475) by [@byCedric](https://github.com/byCedric))
+- Add missing `react` and `react-native` peer dependencies for isolated modules. ([#30475](https://github.com/expo/expo/pull/30475) by [@byCedric](https://github.com/byCedric))
 
 ### 💡 Others
 
 - Removed redundant usage of `EventEmitter` instance. ([#28946](https://github.com/expo/expo/pull/28946) by [@tsapeta](https://github.com/tsapeta))
 - drop unused web `name` property. ([#27437](https://github.com/expo/expo/pull/27437) by [@EvanBacon](https://github.com/EvanBacon))
 - Removed deprecated backward compatible Gradle settings. ([#28083](https://github.com/expo/expo/pull/28083) by [@kudo](https://github.com/kudo))
+- Removed all `NativeModulesProxy occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```
Comment (tsapeta) [inline packages/expo-maps/CHANGELOG.md:21]:
```suggestion
- Removed all `NativeModulesProxy` occurrences. ([#31496](https://github.com/expo/expo/pull/31496) by [@reichhartd](https://github.com/reichhartd))
```

**Review by tsapeta — CHANGES_REQUESTED** [review]
Great, thank you so much! 🙏 I left some changelog suggestions, but in general it looks good.
Recently I had some troubles to move off of `NativeModulesProxy` in `expo-font` tests, but looks like they are passing now – which is great 👍

**Comment by reichhartd** [comment]
You're welcome 🙏
Is it too early to remove the export of `NativeModulesProxy` in the `index.ts` of `expo-modules-core`? Since it would be a breaking change.

**Comment by tsapeta** [comment]
Yeah, I think it's too early. We'll probably need to search over GitHub to see if there are any popular libraries importing it and help them migrate. I would keep it until we entirely remove the legacy modules from the native code.

One more thing we could do is to make it a `Proxy` object and log a warning in its getter, so it's easier to identify places where the deprecated API is used.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**